addons/account/models/account_bank_statement.py

```diff
@@ -83,6 +83,20 @@ def _is_difference_zero(self):
         for bank_stmt in self:
             bank_stmt.is_difference_zero = float_is_zero(bank_stmt.difference, precision_digits=bank_stmt.currency_id.decimal_places)
 
+    @api.depends('date', 'balance_start')
+    def _compute_is_valid_balance_start(self):
+        for bank_stmt in self:
+            domain = [
+                ('journal_id.type', '=', 'bank'),
+                ('date', '<=', bank_stmt.date),
+                ('journal_id', '=', bank_stmt.journal_id.id)
+            ]
+            if bank_stmt._origin:
+                domain = expression.AND([domain, [('id', '<', bank_stmt._origin.id)]])
```

Comment:
I'd specify the order explicitly here, as it's very important for this specific operation